test/controllers/redirect.controller.test.ts

```diff
@@ -1,63 +1,81 @@
+import { INTERNAL_SERVER_ERROR, MOVED_TEMPORARILY } from 'http-status-codes'
 import request from 'supertest'
 import { anything, capture, instance, mock, verify, when } from 'ts-mockito'
 import { ArgCaptor2 } from 'ts-mockito/lib/capture/ArgCaptor'
 import { createApp } from './helpers/application.factory'
 
 import 'app/controllers/redirect.controller'
+import DissolutionApprovalMapper from 'app/mappers/approval/dissolutionApproval.mapper'
 import ApplicationStatus from 'app/models/dto/applicationStatus.enum'
+import DissolutionGetDirector from 'app/models/dto/dissolutionGetDirector'
 import DissolutionGetResponse from 'app/models/dto/dissolutionGetResponse'
 import PaymentStatus from 'app/models/dto/paymentStatus.enum'
+import DissolutionApprovalModel from 'app/models/form/dissolutionApproval.model'
 import DissolutionSession from 'app/models/session/dissolutionSession.model'
 import {
   CERTIFICATE_SIGNED_URI, ENDORSE_COMPANY_CLOSURE_CERTIFICATE_URI,
-  ERROR_URI, PAYMENT_CALLBACK_URI, PAYMENT_URI,
+  PAYMENT_CALLBACK_URI, PAYMENT_URI,
   REDIRECT_GATE_URI,
+  SEARCH_COMPANY_URI,
   SELECT_DIRECTOR_URI,
   VIEW_FINAL_CONFIRMATION_URI,
   WAIT_FOR_OTHERS_TO_SIGN_URI
 } from 'app/paths'
 import DissolutionService from 'app/services/dissolution/dissolution.service'
 import SessionService from 'app/services/session/session.service'
 
-import { generateDissolutionGetResponse } from 'test/fixtures/dissolutionApi.fixtures'
+import { generateApprovalModel, generateDissolutionGetResponse, generateGetDirector } from 'test/fixtures/dissolutionApi.fixtures'
 import { generateDissolutionSession } from 'test/fixtures/session.fixtures'
 
 describe('RedirectController', () => {
 
   let session: SessionService
   let service: DissolutionService
+  let mapper: DissolutionApprovalMapper
 
   const USER_EMAIL = 'myemail@mail.com'
+  const OTHER_USER_EMAIL = 'another@mail.com'
   const TOKEN = 'some-token'
 
   beforeEach(() => {
     session = mock(SessionService)
     service = mock(DissolutionService)
+    mapper = mock(DissolutionApprovalMapper)
 
     when(session.getAccessToken(anything())).thenReturn(TOKEN)
   })
 
+  function initApp(): Application {
+    return createApp(container => {
+      container.rebind(SessionService).toConstantValue(instance(session))
+      container.rebind(DissolutionService).toConstantValue(instance(service))
+      container.rebind(DissolutionApprovalMapper).toConstantValue(instance(mapper))
+    })
+  }
+
   describe('redirect GET request', () => {
+    let dissolutionSession: DissolutionSession
+
+    beforeEach(() => {
+      dissolutionSession = generateDissolutionSession()
+
+      when(session.getDissolutionSession(anything())).thenReturn(dissolutionSession)
+      when(session.getUserEmail(anything())).thenReturn(USER_EMAIL)
+    })
+
     it('should update dissolution session with reference number', async () => {
-      const dissolutionSession: DissolutionSession = generateDissolutionSession()
```

Comment:
there was a lot of duplication across tests so I refactored them into beforeEach blocks